ibis/backends/duckdb/__init__.py

```diff
@@ -1556,8 +1556,12 @@ def _register_in_memory_tables(self, expr: ir.Expr) -> None:
             self._register_in_memory_table(memtable)
 
     def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
-        # only register if we haven't already done so
-        if (name := op.name) not in self.list_tables():
+        name = op.name
+        try:
+            # this handles tables _and_ views
+            self.con.table(name)
+        except (duckdb.CatalogException, duckdb.InvalidInputException):
```

Comment:
The `InvalidInputException` is here for cases where `name` matches a variable in scope that DuckDB will try to magically turn into a table. I don't like it, but that's life 😮‍💨